### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by nilq**
By doing this, Github will format the description properly. Also refactoring code snippets for highlighting.

[Rendered](https://github.com/nilq/minetest/blob/patch-2/README.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,66 +90,67 @@ OS X:
...
 ---------------------
-- Use --help
+- Use `--help`
 
 Compiling on GNU/Linux:
 -----------------------
 
 Install dependencies. Here's an example for Debian/Ubuntu:
-$ sudo apt-get install build-essential libirrlicht-dev cmake libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
+`$ sudo apt-get install build-essential libirrlicht-dev cmake libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev`
```
Comment (juhdanad) [inline README.md]: The command does not start on a new line.

File: README.md [hunk 2]
```diff
@@ -89,66 +90,67 @@ OS X:
...
 
 For Fedora users:
-$ sudo dnf install make automake gcc gcc-c++ kernel-devel cmake libcurl* openal* libvorbis* libXxf86vm-devel libogg-devel freetype-devel mesa-libGL-devel zlib-devel jsoncpp-devel irrlicht-devel bzip2-libs gmp-devel sqlite-devel luajit-devel leveldb-devel ncurses-devel doxygen spatialindex-devel bzip2-devel
+`$ sudo dnf install make automake gcc gcc-c++ kernel-devel cmake libcurl* openal* libvorbis* libXxf86vm-devel libogg-devel freetype-devel mesa-libGL-devel zlib-devel jsoncpp-devel irrlicht-devel bzip2-libs gmp-devel sqlite-devel luajit-devel leveldb-devel ncurses-devel doxygen spatialindex-devel bzip2-devel`
```
Comment (juhdanad) [inline README.md]: Nor this nor the ones after this one start on a new line.

File: README.md [hunk 3]
```diff
@@ -89,66 +90,67 @@ OS X:
...
-$ git clone --depth 1 https://github.com/minetest/minetest_game.git games/minetest_game
+`$ git clone --depth 1 https://github.com/minetest/minetest_game.git games/minetest_game`
 
 Download source, without using git:
-$ wget https://github.com/minetest/minetest/archive/master.tar.gz
-$ tar xf master.tar.gz
-$ cd minetest-master
+`$ wget https://github.com/minetest/minetest/archive/master.tar.gz`
+`$ tar xf master.tar.gz`
+`$ cd minetest-master`
```
Comment (juhdanad) [inline README.md]: These are just next to each other. They should be in separate lines.

File: README.md [hunk 4]
```diff
@@ -89,66 +90,67 @@ OS X:
...
-$ cd games/
-$ wget https://github.com/minetest/minetest_game/archive/master.tar.gz
-$ tar xf master.tar.gz
-$ mv minetest_game-master minetest_game
-$ cd ..
+`$ cd games/`
+`$ wget https://github.com/minetest/minetest_game/archive/master.tar.gz`
+`$ tar xf master.tar.gz`
+`$ mv minetest_game-master minetest_game`
+`$ cd ..`
```
Comment (juhdanad) [inline README.md]: These should be in separate lines.

File: README.md [hunk 5]
```diff
@@ -89,66 +90,67 @@ OS X:
...
 
 Build a version that runs directly from the source directory:
-$ cmake . -DRUN_IN_PLACE=TRUE
-$ make -j <number of processors>
+`$ cmake . -DRUN_IN_PLACE=TRUE`
+`$ make -j <number of processors>`
```
Comment (juhdanad) [inline README.md]: These should be in separate lines.

File: README.md [hunk 6]
```diff
@@ -89,66 +90,67 @@ OS X:
...
 
 Run it:
-$ ./bin/minetest
+`$ ./bin/minetest`
 
 - Use cmake . -LH to see all CMake options and their current state
```
Comment (juhdanad) [inline README.md]: Commands starting with 'cmake' should be formatted as code.

**Comment by paramat** [comment]
:-1: For markdown format, we changed lua_api.txt to markdown and now it's a visual mess in text form, it should not have been changed. The plain text format has priority, we shouldn't need a program to view a cleanly formatted document. The appearence of README on github is of less importance than it's appearence when opened as a plain txt document.

**Comment by rubenwardy** [comment]
> visual mess in text form,

In your opinion

**Comment by rubenwardy** [comment]
> a visual mess in text form

That's because it's a poor translation. 

It's definitely useful for eg: http://rubenwardy.com/minetest_modding_book/lua_api.html

File: README.md [hunk 7]
```diff
@@ -89,66 +90,67 @@ OS X:
 $bin   = Contents/MacOS
 $share = Contents/Resources
 $user  = Contents/User OR ~/Library/Application Support/minetest
+```
 
 World directory
 ----------------
 - Worlds can be found as separate folders in:
-    $user/worlds/
+    `$user/worlds/`
```
Comment (rubenwardy) [inline README.md]: Use indentation rather than \`

File: README.md [hunk 8]
```diff
@@ -71,6 +71,7 @@ Most controls are settable in the configuration file, see the section below.
 
 Paths
 ------
+```
```
Comment (rubenwardy) [inline README.md]: Use a list like the following rather than \`\`\`

* `$bin` - Compiled binaries

File: README.md [hunk 9]
```diff
@@ -89,66 +90,67 @@ OS X:
...
 ----------------
 - Worlds can be found as separate folders in:
-    $user/worlds/
+    `$user/worlds/`
 
 Configuration file:
 -------------------
 - Default location:
-    $user/minetest.conf
+    `$user/minetest.conf`
```
Comment (rubenwardy) [inline README.md]: Use indentation rather than `

File: README.md (hunk 6 above)
Comment (rubenwardy) [inline README.md]: this is basically the only place you should use ` - for inline code

**Comment by est31** [comment]
:+1: to making it md, but obviously it shouldn't look broken when rendered.

**Comment by rubenwardy** [comment]
#6244

**Pull request closed without merging**